backend/layers/common/entities.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import List
+
+
+@dataclass
+class DatasetStatus:
+    status: str  # TODO: use an enum
+
+
+@dataclass
+class DatasetArtifact:
+    type: str
+    uri: str
+
+
+@dataclass
+class DatasetMetadata:
+    organism: str
+    tissue: str
+    assay: str
+    disease: str
+    sex: str
+    self_reported_ethnicity: str
+    development_stage: str
+    cell_type: str
+    cell_count: int
+
+
+@dataclass
+class Dataset:
+    id: str
+    status: DatasetStatus
+    metadata: DatasetMetadata
+    artifacts: List[DatasetArtifact]
+
+
+@dataclass
+class CollectionMetadata:
+    name: str
+    description: str
+    owner: str
+    contact_name: str
+    contact_email: str
+    links: List[dict]  # TODO: use a dataclass
+
+
+@dataclass
+class Collection:
+    id: str
+    metadata: CollectionMetadata
+    publisher_metadata: dict # TODO: use a dataclass
+    datasets: List[Dataset]
```

Comment:
The latter is definitely true (in fact, that's how the mock is implemented - I need to set up pylance to attempt some static typing and catch those errors !).

For the former, it is mostly a coding design choice - those 4 fields are just grouped under a `Collection` class for ease of use. When you create a version, you want to clone those fields verbatim and copy them to the new version (since initially a version/revision is identical to the published). Grouping them allows for an easier copy. Note that this `Collection` doesn't have anything to do with the `Collections` table we designed, which will only be a mapping between ids.